### DISCUSSION TITLE
How to add custom keyword to Babel parser?

### DISCUSSION THREAD

**Discussion by skeggse** [babel/babel#12069] [General]
## Feature Request

- [x] I would like to work on this feature!

**Describe the solution you'd like**

I'd like to add support for a `defer` keyword to the babel parser, default disabled. This is well outside the scope of any standards proposals, so I'm not sure what the policy is concerning such features in the babel parser. I'd love to be able to write a babel plugin to implement golang-style `defer` functionality in babel, and it'd be neat to be able to use an actual `defer` statement. It should be a pretty small lift, considering it'd only be valid at the start of a Statement, and wouldn't require the addition of a new reserved keyword (similar to `async`).

**Describe alternatives you've considered**.

It's entirely possible to simply implement this without the keyword. Making `defer <id>` a valid statement is just style points. [Prior art](https://github.com/steelbrain/babel-plugin-defer-logic). Happy to accept this is "out of scope" for babel's parser.

**Documentation, Adoption, Migration Strategy**
If you can, explain how users will be able to use this and how it might be documented. Maybe a mock-up?

`@babel/plugin-syntax-defer`, or something similar.

[a comment by an automated account is not shown]

**Comment by nicolo-ribaudo** [comment]
Yes, this is definitely out-of-scope. However, if you want to fork `@babel/parser`, you can then pass it to `@babel/core` using your `syntax-defer` plugin:

```js
export function myPlugin() {
  return {
    parserOverride(code, opts) {
      return parseWithCustomFork(code, opts);
    },
  };
}
```

You can see how `void` or `delete` are implemented in `@babel/parser`, and implement `defer` in the same way.
